console/src/main/java/org/eclipse/rdf4j/console/command/Verify.java

```diff
@@ -118,14 +152,76 @@ public void execute(String... tokens) {
 	}
 
 	/**
-	 * Parse URL or path to local file. Only the second (index 1) token will be parsed, files will be prefixed with
-	 * "file:" scheme
+	 * Validate an RDF data source using a SHACL file or URL, writing the report to a file
 	 * 
-	 * @param tokens array of tokens to be parsed
+	 * @param dataPath  file or URL of the data to be validated
+	 * @param shaclPath file or URL of the SHACL
+	 * @parem reportFile file to write validation report to
+	 */
+	private void shacl(String dataPath, String shaclPath, String reportFile) {
+		ShaclSail sail = new ShaclSail(new MemoryStore());
+		SailRepository repo = new SailRepository(sail);
+		repo.init();
+
+		// load shapes first
+		boolean loaded = false;
+		try {
+			consoleIO.writeln("Loading shapes from " + shaclPath);
+
+			URL shaclURL = new URL(shaclPath);
+			RDFFormat format = Rio.getParserFormatForFileName(shaclPath).orElseThrow(Rio.unsupportedFormat(shaclPath));
+
+			try (SailRepositoryConnection conn = repo.getConnection()) {
+				conn.add(shaclURL, "", format, RDF4J.SHACL_SHAPE_GRAPH);
+			}
+			loaded = true;
+		} catch (MalformedURLException e) {
+			consoleIO.writeError("Malformed URL: " + shaclPath);
+		} catch (IOException e) {
+			consoleIO.writeError("Failed to load shacl shapes: " + e.getMessage());
+		}
+
+		if (!loaded) {
+			consoleIO.writeError("No shapes found");
+			repo.shutDown();
+			return;
+		}
+
+		sail.setParallelValidation(false);
+		sail.setRdfsSubClassReasoning(false);
```

Comment:
The SHACL spec supports rdfs:subClassOf, which can be disabled in the ShaclSail with the command above. Probably shouldn't be disabled though, unless this is what the user wants.

Reply:
ok, good point